File: 设计模式/程序/22-StatePattern/StatePattern(C#)/Subject.cs
using System;

namespace StatePatternApp
{
    class Subject
    {
        public Subject(State state)
        {
            this.state = state;
        }
        public void SetState(State state)
        {
            this.state = state;
        }

        public void Operation1(string message)
        {
            this.state.DoOperation1(message);
        }
        public void Operation2(string message)
        {
            this.state.DoOperation2(message);
        }
        public void Operation3(string message)
        {
            this.state.DoOperation3(message);
        }

        private State state;
    }
}

File: 设计模式/程序/22-StatePattern/StatePattern(python)/Subject.py
class Subject:
    def __init__(self, state):
        self.state = state

    def SetState(self, state):
        self.state = state

    def Operation1(self, message):
        self.state.DoOperation1(message)

    def Operation2(self, message):
        self.state.DoOperation2(message)

    def Operation3(self, message):
        self.state.DoOperation3(message)